addons/popochiu/editor/factories/factory_base_popochiu_obj.gd

```diff
@@ -0,0 +1,199 @@
+extends RefCounted
+
+const BASE_STATE_TEMPLATE := 'res://addons/popochiu/engine/templates/%s_state_template.gd'
+const BASE_SCRIPT_TEMPLATE := 'res://addons/popochiu/engine/templates/%s_template.gd'
+const BASE_SCENE_PATH := 'res://addons/popochiu/engine/objects/%s/popochiu_%s.tscn'
+const EMPTY_SCRIPT := 'res://addons/popochiu/engine/templates/empty_script_template.gd'
+
+const Constants := preload('res://addons/popochiu/popochiu_resources.gd')
+const MainDock := preload('res://addons/popochiu/editor/main_dock/popochiu_dock.gd')
+const PopochiuObjectRow := preload('res://addons/popochiu/editor/main_dock/object_row/popochiu_object_row.gd')
+
+var _main_dock: Panel = null
+var _ei: EditorInterface
+
+# The following variables are setup on creation
+# Names variants and name parameter passed to
+# the create method.
+var _obj_path_template := '' # always set by child class
+var _obj_snake_name := ''
+var _obj_pascal_name := ''
+var _obj_path_base := ''
+var _obj_path_script := ''
+var _obj_path_state = ''
+var _obj_path_resource = ''
+var _obj_path_scene = ''
+# The following variables are setup by the sub-class constructor
+# to define the type of object to be processed
+# TODO: reduce this to just "type", too much redundancy
+var _obj_type := -1
+var _obj_type_label := ''
+var _obj_type_target := ''
+# The following variables are references to the elements
+# generated for the creation of the new Popochiu object,
+# such as resources, scenes, scripts, state scripts, etc
+var _obj: Node # Can be a scene or a simple node in the tree
+var _obj_state_resource: Resource
+var _obj_script: Resource
+var _obj_dock_row: PopochiuObjectRow
```

Comment:
It seems redundant and in fact, in retrospect, it is. I used it in the beginning because I didn't know if more variables had to be added that were unrelated to the new to-be-created object.

I can't decide if this still holds some value (like in the future we may evolve those classes to have members representing properties of the factory, not the object, or not. Also, since the whole thing is now encapsulated, the naming redundancy is just in a single class.

Changing it is not a problem (I have my tools for that :D). If you prefer I can get rid of the prefixes.

Reply:
I think the code would feel cleaner if you remove those prefixes.

Reply:
OK, I'll do asap.

Reply:
Done and tested, it seems everything works as intended!